src/smt/theory_lra.cpp

```diff
@@ -1059,12 +1059,54 @@ class theory_lra::imp {
         return true;
     }
 
+    svector<std::pair<theory_var, theory_var>> m_diseqs;
+    unsigned m_diseqs_qhead = 0;
+
     void new_diseq_eh(theory_var v1, theory_var v2) {
```

Comment:
we would actually want the same updates in sat/smt/arith_solver.cpp

Reply:
Do you mean adding the chunk below ta arith_solver.cpp? It doesn't compile...
` 
 // actually have to go over entire stack to ensure diseqs are respected after mutations
    bool delayed_diseqs() {
        if (m_diseqs_qhead == m_diseqs.size())
            return false;
        ctx().push_trail(value_trail(m_diseqs_qhead));
        bool has_eq = false;
        while (m_diseqs_qhead < m_diseqs.size()) {
            auto [v1,v2] = m_diseqs[m_diseqs_qhead];
            if (is_eq(v1, v2)) {
                m_arith_eq_adapter.new_diseq_eh(v1, v2);
                has_eq = true;
            }
            ++m_diseqs_qhead;
        }
        return has_eq;
    }

    svector<std::pair<theory_var, theory_var>> m_deqs;
    unsigned m_eqs_qhead = 0;

    // actually have to go over entire stack to ensure eqs are respected after mutations
    bool delayed_eqs() {
        if (m_eqs_qhead == m_deqs.size())
            return false;
        ctx().push_trail(value_trail(m_eqs_qhead));
        bool has_eq = false;
        while (m_eqs_qhead < m_deqs.size()) {
            auto [v1,v2] = m_deqs[m_eqs_qhead];
            if (!is_eq(v1, v2)) {
                //verbose_stream() << "bad diseq " << m_diseqs_qhead << "\n";
                m_arith_eq_adapter.new_eq_eh(v1, v2);
                has_eq = true;
            }
            ++m_eqs_qhead;
        }
        return has_eq;
    }
`